lib/main.dart

```diff
@@ -1,6 +1,38 @@
 import 'package:flutter/material.dart';
+import 'dart:convert';
+import 'dart:async';
+import 'package:http/http.dart' as http;
+import 'package:flutter/foundation.dart';
+import 'catData.dart';
 
 void main() {
-  runApp(MyApp());
+  runApp(MainPage());
```

Comment:
Think about names for the classes as "what thing this class do"
for example a good name for app class is CatsAndDogsApp.